share/shwap/p2p/bitswap/getter.go

```diff
@@ -0,0 +1,172 @@
+package bitswap
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ipfs/boxo/blockstore"
+	"github.com/ipfs/boxo/exchange"
+
+	"github.com/celestiaorg/celestia-app/pkg/wrapper"
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/header"
+	"github.com/celestiaorg/celestia-node/share"
+)
+
+// Getter implements share.Getter.
+type Getter struct {
+	exchange exchange.SessionExchange
+	bstore   blockstore.Blockstore
+	session  exchange.Fetcher
+	cancel   context.CancelFunc
+}
+
+// NewGetter constructs a new Getter.
+func NewGetter(exchange exchange.SessionExchange, bstore blockstore.Blockstore) *Getter {
+	return &Getter{exchange: exchange, bstore: bstore}
+}
+
+// Start kicks off internal fetching session.
+func (g *Getter) Start() {
+	ctx, cancel := context.WithCancel(context.Background())
+	g.session = g.exchange.NewSession(ctx)
+	g.cancel = cancel
```

Comment:
Would it be possible for Fetch to read from blockstore before requesting from network? It seems like it will always network exchange as fetcher here, when we would want to have local lookup from blockstore first.

Reply:
I don't see when we would want that. We have store getter for that

Reply:
After failed reconstruction, next one (retry) should be able to read samples from blockstore that prevoius attempt has saved.

Reply:
This should be part of `Retriever` logic. This way we avoid hitting store(and disk) for when we don't need it

Reply:
Do I understand it that the blockstore provided to Fetch will be used for storing samples by Lights Nodes? Perhaps we can also store samples onside of Getter for LN. For example in Availability, same way Full nodes do it outside of getter

Reply:
This is likely to happen eventually